src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentBag.cs

```diff
@@ -973,7 +973,7 @@ internal bool TrySteal([MaybeNullWhen(false)] out T result, bool take)
                         // is in progress, as add operations need to accurately count transitions
                         // from empty to non-empty, and they can only do that if there are no concurrent
                         // steal operations happening at the time.
-                        if (head < _tailIndex - 1 && _currentOp != (int)Operation.Add)
+                        if ((head - (_tailIndex - 2) >= 0) && _currentOp != (int)Operation.Add)
```

Comment:
I think you forgot to fix that part of the condition. Should be `_currentOp == (int)Operation.Add`

Reply:
Thank goodness I wrote comments ;)